flexmeasures/ui/templates/crud/asset.html

```diff
@@ -369,6 +483,438 @@ <h3>All child assets for {{ asset.name }}</h3>
+    }
+
+    // Fetch Asset Details
+    async function getAsset(assetId){
+        const cacheKey = `asset_${assetId}`;
+        const cachedData = localStorage.getItem(cacheKey);
+
+        if (cachedData) {
+            return JSON.parse(cachedData);
+        }
+
+        const apiUrl = apiBasePath + "/api/v3_0/assets/" + assetId;
+        const response = await fetch(apiUrl);
+        const asset = await response.json();
+
+        localStorage.setItem(cacheKey, JSON.stringify(asset));
+
+        return asset;
+    }
+
+    // Fetch Sensor Details
+    async function getSensor(id) {
+        const cacheKey = `sensor_${id}`;
+        const cachedData = localStorage.getItem(cacheKey);
+
+        if (cachedData) {
+            return JSON.parse(cachedData);
+        }
+
+        const apiUrl = apiBasePath + "/api/v3_0/sensors/" + id;
+        const response = await fetch(apiUrl);
+        const sensor = await response.json();
+        
+        localStorage.setItem(cacheKey, JSON.stringify(sensor));
+        
+        return sensor;
+    }
+
+    // highlight selected graph 
+    async function selectGraph(graphIndex) {
+        if (graphIndex !== undefined){
+            savedGraphIndex = graphIndex;
+            selectedGraphTitle = data[graphIndex]?.title;
+            // check if graphIndex still exists(This is because this function is called when the removed button is clicked as well)
+            if (data[graphIndex]) {
+                renderApiSensors(cachedFilteredSensors, graphIndex);
+            } else {
+                renderApiSensors(cachedFilteredSensors);
+            }
+        } else {
+            savedGraphIndex = undefined;
+            selectedGraphTitle = undefined;
+            renderGraphCards();
+        }
+    }
+
+    // Function to render the graph cards
+    async function renderGraphCards() {
+        const graphList = document.getElementById("graphList");
+        graphList.innerHTML = ""; 
+        const newData = [];
+
+        if (data.length === 0) {
+            return;
+        }
+
+        for (const [index, item] of data.entries()) {
+            const col = document.createElement("div");
+            col.classList.add("col-12", "mb-1");
+
+            const sensorsUnits = [];
+            const newItem = {
+                title: item.title ?? "No Title", 
+                sensors: item.sensors ?? (Array.isArray(item) ? item : [item]),  
+            }
+            newData.push(newItem);
+
+            const sensorsContent = await Promise.all(
+                newItem.sensors.map(async (sensor, sensorIndex) => {
+                    const sensorData = await getSensor(sensor);
+                    const Asset = await getAsset(sensorData.generic_asset_id);
+                    const Account = await getAccount(Asset.account_id);
+                    sensorsUnits.push(sensorData.unit);
+            
+                    return `
+                    <div class="p-1 mb-3 border-bottom border-secondary">
```

Comment:
I find it visually a bit confusing that this "X" icon is top of the `span`. It sits almost within the text, so the second row of text can be mistaken as a different thing, especially if there is only one sensor.

I could improve it by adding it as first element to the `div` and adding `float: right;` - but maybe you have a better idea.

Reply:
Ok, i can work on that, there are some Bootstrap classes I could use